### PR TITLE
Fixed potential deadlock when a subscription is disposed from a transaction handler

### PR DESCRIPTION
**Pull request by IharBury**

### CODE REVIEW

**Comment by IharBury** [comment]
@dennisdoomen, I don't have permissions for this repo for some reason

**Comment by IharBury** [comment]
Hmm, I think we can get a problem with this approach when disposing the adapter itself. I'll add some more commits...

**Comment by IharBury** [comment]
Now it should be fine

File: Tests/LiquidProjections.PollingEventStore.Specs/NEventStoreAdapterSpecs.cs [hunk 1]
```diff
@@ -418,5 +418,47 @@ public void Then_the_second_subscription_should_not_hang()
...
+                            info.Subscription.Dispose();
+                            disposed.Set();
+                            return Task.FromResult(0);
+                        }
+                    },
+                    "someId"));
+            }
+
+            [Fact]
+            public void Then_it_should_stop()
```
Comment (dennisdoomen) [inline Tests/LiquidProjections.PollingEventStore.Specs/NEventStoreAdapterSpecs.cs]: Suggestion: `Then_it_should_cancel_the_subscription_asynchronously`

File: Tests/LiquidProjections.PollingEventStore.Specs/NEventStoreAdapterSpecs.cs [hunk 2]
```diff
@@ -418,5 +418,47 @@ public void Then_the_second_subscription_should_not_hang()
                 }
             }
         }
+        
+        public class When_disposing_subscription_from_a_transaction_handler : GivenSubject<PollingEventStoreAdapter>
```
Comment (dennisdoomen) [inline Tests/LiquidProjections.PollingEventStore.Specs/NEventStoreAdapterSpecs.cs]: Suggestion: `When_the_subscriber_cancels_the_subscription_from_inside_its_transaction_handler`

**Review by dennisdoomen — APPROVED** [review]

**Comment by IharBury** [comment]
Renamed

**Pull request merged**